paasta_tools/api/views/autoscaler.py

```diff
@@ -20,8 +20,51 @@
 
 from paasta_tools.api import settings
 from paasta_tools.api.views.exception import ApiFailure
+from paasta_tools.kubernetes_tools import load_kubernetes_service_config
 from paasta_tools.marathon_tools import load_marathon_service_config
-from paasta_tools.marathon_tools import set_instances_for_marathon_service
+from paasta_tools.utils import NoConfigurationForServiceError
+from paasta_tools.utils import validate_service_instance
+
+
+def get_instance_type(service, instance, cluster, soa_dir):
+    try:
+        return validate_service_instance(service, instance, cluster, soa_dir)
+    except NoConfigurationForServiceError:
+        error_message = (
+                "Deployment key %s not found. Try to execute the corresponding pipeline if it's a fresh instance"
+                % ".".join([settings.cluster, instance])
+        )
+        raise ApiFailure(error_message, 404)
+    except Exception as e:
+        raise ApiFailure(e, 500)
+
+
+def get_service_config(instance_type, service, instance, cluster, soa_dir):
+    try:
+        if instance_type == 'marathon':
```

Comment:
Nit: use a map like https://github.com/Yelp/paasta/blob/master/paasta_tools/kubernetes_tools.py#L140 
Map "kubernetes"/"marathon" to functions.